adapters/metax/metax.go

```diff
@@ -0,0 +1,196 @@
+		})
+	}
+
+	return reqDatas, errs
+}
+
+func (a *adapter) MakeBids(bidReq *openrtb2.BidRequest, reqData *adapters.RequestData, respData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if adapters.IsResponseStatusCodeNoContent(respData) {
+		return nil, nil
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(respData); err != nil {
+		return nil, []error{err}
+	}
+
+	var bidResp openrtb2.BidResponse
+	if err := json.Unmarshal(respData.Body, &bidResp); err != nil {
+		return nil, []error{err}
+	}
+
+	// additional no content check
+	if len(bidResp.SeatBid) == 0 || len(bidResp.SeatBid[0].Bid) == 0 {
+		return nil, nil
+	}
+
+	resp := adapters.NewBidderResponseWithBidsCapacity(len(bidResp.SeatBid[0].Bid))
+	if len(bidResp.Cur) != 0 {
+		resp.Currency = bidResp.Cur
+	}
+	for _, sb := range bidResp.SeatBid {
+		for i := range sb.Bid {
+			bid := &sb.Bid[i]
+			bidType, err := getBidType(bid)
+			if err != nil {
+				return nil, []error{err}
+			}
+			resp.Bids = append(resp.Bids, &adapters.TypedBid{
+				Bid:     bid,
+				BidType: bidType,
+			})
+		}
+	}
+	return resp, nil
+}
+
+func (a *adapter) getEndpoint(ext *openrtb_ext.ExtImpMetaX) (string, error) {
+	params := macros.EndpointTemplateParams{
+		PublisherID: strconv.Itoa(ext.PublisherID),
+		AdUnit:      strconv.Itoa(ext.Adunit),
+	}
+	return macros.ResolveMacros(a.template, params)
+}
+
+func parseBidderExt(imp *openrtb2.Imp) (*openrtb_ext.ExtImpMetaX, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, err
+	}
+
+	var metaxExt openrtb_ext.ExtImpMetaX
+	if err := json.Unmarshal(bidderExt.Bidder, &metaxExt); err != nil {
+		return nil, errors.New("Wrong MetaX bidder ext")
+	}
+
+	return &metaxExt, nil
+}
+
+func preprocessImp(imp *openrtb2.Imp) error {
+	if imp == nil {
+		return errors.New("imp is nil")
+	}
+
+	if imp.Banner != nil {
+		imp.Banner = assignBannerSize(imp.Banner)
+	}
+
+	return nil
+}
+
+func assignBannerSize(banner *openrtb2.Banner) *openrtb2.Banner {
+	if banner.W != nil && banner.H != nil {
+		return banner
+	}
+
+	if len(banner.Format) == 0 {
+		return banner
+	}
+
+	return assignBannerWidthAndHeight(banner, banner.Format[0].W, banner.Format[0].H)
+}
+
+func assignBannerWidthAndHeight(banner *openrtb2.Banner, w, h int64) *openrtb2.Banner {
+	bannerCopy := *banner
+	bannerCopy.W = ptrutil.ToPtr(w)
+	bannerCopy.H = ptrutil.ToPtr(h)
+	return &bannerCopy
+}
+
+func getBidType(bid *openrtb2.Bid) (openrtb_ext.BidType, error) {
+	switch bid.MType {
+	case openrtb2.MarkupBanner:
+		return openrtb_ext.BidTypeBanner, nil
+	case openrtb2.MarkupVideo:
+		return openrtb_ext.BidTypeVideo, nil
+	case openrtb2.MarkupNative:
+		return openrtb_ext.BidTypeNative, nil
+	case openrtb2.MarkupAudio:
+		return openrtb_ext.BidTypeAudio, nil
+	default:
+		return "", &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unsupported MType %d", bid.MType),
+		}
+	}
```

Comment:
please add test for Native and Audio
![image](https://github.com/user-attachments/assets/139efdf4-6d91-4196-b08d-082ea2d27569)

Reply:
hi przemkaczmarek,
Do you means bid.MType test? It seems you take screenshot of old commit, we have the MType tests in metaxtest/exemplary/(simple-app-audio.json, simple-app-native.json, etc.)
Please give me more details so I can fix it

Reply:
It was old commit, close comment